source/parse.h

```diff
@@ -8647,6 +8692,15 @@ class parser
         auto n = std::make_unique<translation_unit_node>();
         for (auto d = declaration(); d; d = declaration()) {
             n->declarations.push_back( std::move(d) );
+
+            // Add declarations generated by metafunctions.
+            if (!metafunction_declarations.empty()) {
+                for(auto& md : metafunction_declarations) {
+                    n->declarations.push_back( std::move(md));
+                }
+
+                metafunction_declarations.clear();
+            }
```

Comment:
The if statement is redundant.
```suggestion
            for(auto& md : metafunction_declarations) {
                n->declarations.push_back( std::move(md));
            }

            metafunction_declarations.clear();
```